src/terminal.c

```diff
@@ -268,6 +278,49 @@ terminal_setflags(EditLine *el)
 libedit_private int
 terminal_init(EditLine *el)
 {
+	char tinfo_so[TC_BUFSIZE];
+	int major_version;
+
+	for (major_version = NCURSES_HIGH_VERSION; major_version >= NCURSES_LOW_VERSION; major_version--){
+		sprintf (tinfo_so, "libtinfo.so.%d", major_version);
+		if ((dl_handle = dlopen (tinfo_so, RTLD_LAZY)) != NULL)
+			break;
+	}
+
+	if (dl_handle == NULL){
+		fprintf (stderr, "ERROR: Cannot load ncurses library.\n");
+		goto fail1;
```

Comment:
If an error is returned when the library cannot be loaded or the corresponding function is not found, segfault may occur due to incorrect error handling  at the calling function.
It seems better to exit.

Reply:
I hadn't considered that.
You right.
Errors related to **dlopen ()** and **dlsym ()** cannot be recovered, and result in coredump.
So _exit should be used.